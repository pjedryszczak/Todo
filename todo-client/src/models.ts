export interface Todo {
        id?: number,
        content: string,
        todoListId: number,
        checked: boolean
    }
export interface TodoList {
        id?: number,
        userId: number,
        title: string
    }   

 