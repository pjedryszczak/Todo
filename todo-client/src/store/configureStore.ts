import {createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";
import createSagaMiddleware from "redux-saga";
import sagas from "./sagas";
import { reducer } from "./reducer";

export default function configureStore(initialState: any, history: any) {
    const composeEnhancers = composeWithDevTools({});
    const reduxSagaMiddleware = createSagaMiddleware();

    const middleware = [reduxSagaMiddleware, thunk ];
    const applyMiddlewareFunc = applyMiddleware(...middleware);

    let store: any;
    if(process.env.NODE_ENV === 'production') {
        store = createStore(reducer, applyMiddlewareFunc);
    }else{
        store = createStore(reducer,initialState, composeEnhancers(applyMiddlewareFunc));
    }

    reduxSagaMiddleware.run(sagas);

    return store;
}