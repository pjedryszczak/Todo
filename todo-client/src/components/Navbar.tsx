import React, { Component } from 'react';
import {  RouteComponentProps } from 'react-router-dom';
import { BiBookBookmark } from 'react-icons/bi'
import { logout } from '../store/actions';
import { TodoAppState } from '../store/reducer';
import { User } from '../models';
import { connect } from 'react-redux';
interface DispatchProps {
  logout: typeof logout
}
interface StoreState {
  loggedIn: boolean,
  user: User
}
interface LocalState {
  loggedIn: boolean,
  user: User
}
type Props = DispatchProps & StoreState & RouteComponentProps;
class Navbar extends Component<Props, LocalState> {
  state: LocalState={
    loggedIn: false,
    user: this.props.user
  }
  static getDerivedStateFromProps(props: StoreState, state: LocalState) {
        
    return {
      loggedIn: props.loggedIn,
      user: props.user
    };          
}
  logout = () => {    
    this.props.logout(this.props.history);
  }
      render(){
      return (
        <nav className="nav-wrapper blue darken-3">
            <div className="cointainer">
                <a href="/" className="brand-logo" ><BiBookBookmark/> Todo App</a>
                <ul className="right">                     
                    {this.props.loggedIn ? <a href="/" onClick={(e) => this.logout()}>Log Out</a> : <></>}
                </ul>
                <ul className="right">
                {this.props.loggedIn ? <span>Hello, {this.props.user.firstName} :)</span> : <></>}
                </ul>
            </div>
        </nav>
      );
    }
    
  }
  function mapStateToProps(state: TodoAppState, props: any){
    return {
      loggedIn: state.loggedIn,
      user: state.user
     }
}
  const container = connect(
    mapStateToProps,
    {
    logout
    }
  )(Navbar)

export default container;