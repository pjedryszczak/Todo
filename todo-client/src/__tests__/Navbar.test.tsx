import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import Navbar from '../components/Navbar';
import { Provider } from 'react-redux';
import { User } from '../models';

const mockStore = configureStore([]);
describe('Navbar', () => {
    let store: any;
    let component: renderer.ReactTestRenderer; 
   const user: User ={
    firstName: 'TestName',
    id: 0,
    username: 'testname',
    token: ''
   }
    beforeEach(() => {
      store = mockStore({
        loggedIn: true,
        user: user
      });
      store.dispatch = jest.fn();

      component = renderer.create(
        <Provider store={store}>
            <Navbar />
        </Provider>);
    });

    it('it renders with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
  });