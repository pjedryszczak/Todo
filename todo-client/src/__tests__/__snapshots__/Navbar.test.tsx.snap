// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar it renders with given state from Redux store 1`] = `
<nav
  className="nav-wrapper blue darken-3"
>
  <div
    className="cointainer"
  >
    <a
      className="brand-logo"
      href="/"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19,2.01H6c-1.206,0-3,0.799-3,3v3v6v3v2c0,2.201,1.794,3,3,3h15v-2H6.012C5.55,19.998,5,19.815,5,19.01 c0-0.101,0.009-0.191,0.024-0.273c0.112-0.575,0.583-0.717,0.987-0.727H20c0.018,0,0.031-0.009,0.049-0.01H21v-0.99V15V4.01 C21,2.907,20.103,2.01,19,2.01z M19,16.01H5v-2v-6v-3c0-0.806,0.55-0.988,1-1h7v7l2-1l2,1v-7h2V15V16.01z"
        />
      </svg>
       Todo App
    </a>
    <ul
      className="right"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Log Out
      </a>
    </ul>
    <ul
      className="right"
    >
      <span>
        Hello, 
        TestName
         :)
      </span>
    </ul>
  </div>
</nav>
`;
